cli/lock.ml

```diff
@@ -170,6 +170,24 @@ let root_pin_depends local_opam_files =
     local_opam_files []
   |> D.Pin_depends.sort_uniq
 
+let get_local_pins source_config =
+  match (source_config : D.Source_opam_config.t) with
+  | { repositories = Some _; _ } ->
+      (* ignore local pins with using x-opam-monorepo-repositories *)
+      []
+  | _ ->
+      OpamGlobalState.with_ `Lock_none @@ fun global_state ->
+      OpamSwitchState.with_ `Lock_none global_state @@ fun switch_state ->
+      OpamPinned.packages switch_state
+      |> OpamPackage.Set.to_seq
+      |> Seq.filter_map (fun pkg ->
+             match OpamSwitchState.url switch_state pkg with
+             | None ->
+                 (* ignore version-pinned packages *)
+                 None
```

Comment:
Maybe it would make more sense to make this into a named function `is_pinned` then it would be clearer that checking for an empty URL means a pinned package (or actually, maybe the OPAM libraries should expose this in some more obviously correct way instead of us trying to interpret what OPAM does).